### ISSUE TITLE
The extension slows down Threads considerably

### ISSUE THREAD

**Issue by eugenioperea** [TheKonka/instagram-download-browser-extension#31]
Browser(Chrome/Firefox): Firefox on Ubuntu Linux
Extension Version: 2.1.7
Reproduction of links: Any threaded post on Threads.net

For the past couple of weeks I've noticed that Threads is unbearably slow. When I open a post to see the subsequent posts they can take up to 20 seconds to load. I disabled all the extension on my system and started testing each one. This one turned out to be the culprit.

**Comment by TheKonka** [comment]
![image](https://github.com/TheKonka/instagram-download-browser-extension/assets/22173084/abe87459-b4fc-4ec7-b4f0-f737d212bda1)

Really sorry for this,  the next version will support disabling the threads download individually.

**Comment by eugenioperea** [comment]
You have absolutely nothing to apologize for. Your work is great, and I appreciate the effort you put in. Thank you for responding so promptly. I'd already disabled that option earlier today, and all is good. I'll wait for the next version.

**Comment by Cpt-KeksMonster** [comment]
Yes, unfortunately, the add-on slows down my Firefox even when the toggle switch for Threads is turned off.

Threads only works at normal speed when the add-on is completely disabled. Additionally, the add-on is currently resource-hungry and causes the entire system to lag (i7 CPU 10th Gen, 32 GB RAM, PCIe SSD). When the add-on is active, Firefox's RAM usage briefly spikes. At its peak, I saw 7 GB instead of just over 3 GB without the add-on and the use of Threads.

It's such a shame because I love the add-on, especially for Instagram, but for now, I'll keep it disabled and only enable it when I want to download something. Hopefully, there's potential for further improvements.

**Thanks for the hard work and for providing the add-on!**

**Comment by eugenioperea** [comment]
I'm no longer seeing the CPU rise with the extension. I re-enabled all permissions a few weeks ago, and it's been fine.

**Comment by Nissemus** [comment]
Haven't had this problem on Windows - maybe it's a Linux thing?

**Comment by eugenioperea** [comment]
Hi, the problem is back. RAM usage goes through the roof when a Threads tab is opened. I saw 3 GB a minute ago. I followed the instructions [here](https://support.mozilla.org/bm/questions/1440662) and indeed it was this extension.

**Comment by eugenioperea** [comment]
I've just updated to the new version. Problem persists, and it's even worse now. It shot up to 4 GB in less than 15 seconds. @TheKonka are you still keeping up with this issue? It feel like I'm shouting into the void.

**Comment by TheKonka** [comment]
> I've just updated to the new version. Problem persists, and it's even worse now. It shot up to 4 GB in less than 15 seconds. [@TheKonka](https://github.com/TheKonka) are you still keeping up with this issue? It feels like I'm shouting into the void.

Sorry, I am aware of this issue, but I cannot think of a good solution at the moment.

**Comment by eugenioperea** [comment]
Got it. Thanks.
